backend/googlephotos/googlephotos.go

```diff
@@ -654,24 +654,38 @@ func (f *Fs) listDir(ctx context.Context, prefix string, filter api.SearchFilter
 	if err != nil {
 		return nil, err
 	}
-	// Dedupe the file names
-	dupes := map[string]int{}
+	// Count occurrences of IDs and filenames
+	filenameCount := map[string]int{}
+	seen_ids := map[string]bool{}
 	for _, entry := range entries {
 		o, ok := entry.(*Object)
 		if ok {
-			dupes[o.remote]++
+			if seen_ids[o.id] {
+				continue
+			}
+			seen_ids[o.id] = true
+			filenameCount[o.remote]++
 		}
 	}
+	seen_ids = map[string]bool{}
+	newEntries := make([]fs.DirEntry, 0, len(entries))
 	for _, entry := range entries {
 		o, ok := entry.(*Object)
-		if ok {
-			duplicated := dupes[o.remote] > 1
-			if duplicated || o.remote == "" {
-				o.remote = addFileID(o.remote, o.id)
-			}
+		if !ok {
+			newEntries = append(newEntries, entry)
```

Comment:
Not sure if necessary to append in case ok is false or just continue is enough, but left it here to better match the original entries' contents. Please let me know if this should be safe to remove and we can simply `continue`.